provider/provider.go

```diff
@@ -71,19 +78,70 @@ func (p *Provider) GetProviderSchema(_ context.Context, _ *cqproto.GetProviderSc
 	}, nil
 }
 
-func (p *Provider) GetProviderConfig(_ context.Context, _ *cqproto.GetProviderConfigRequest) (*cqproto.GetProviderConfigResponse, error) {
-	providerConfig := p.Config()
+func (p *Provider) GetProviderConfig(_ context.Context, req *cqproto.GetProviderConfigRequest) (*cqproto.GetProviderConfigResponse, error) {
+	providerConfig := p.Config(req.Format)
 	if err := defaults.Set(providerConfig); err != nil {
 		return &cqproto.GetProviderConfigResponse{}, err
 	}
-	data := fmt.Sprintf(`
+	switch providerConfig.Format() {
+	case cqproto.ConfigHCL:
+		data := fmt.Sprintf(`
 		provider "%s" {
 			%s
 			// list of resources to fetch
 			resources = %s
-		}`, p.Name, p.Config().Example(), helpers.FormatSlice(funk.Keys(p.ResourceMap).([]string)))
+		}`, p.Name, providerConfig.Example(), helpers.FormatSlice(funk.Keys(p.ResourceMap).([]string)))
+
+		return &cqproto.GetProviderConfigResponse{
+			Config: hclwrite.Format([]byte(data)),
+			Format: cqproto.ConfigHCL,
+		}, nil
+	case cqproto.ConfigYAML:
+		var data ProviderConfiguration
+		if err := yaml.Unmarshal([]byte(providerConfig.Example()), &data.Inline); err != nil {
```

Comment:
why the unmarshal/marshal is necessary here? why not just send the []byte/string as is?

Reply:
because we add resource list to the bottom in line 105..130